crates/nargo_cli/build.rs

```diff
@@ -119,51 +84,81 @@ fn compile_success_empty_{test_name}() {{
             "#,
-            test_dir = test_dir.display(),
-        )
-        .expect("Could not write templated test file.");
-    }
-}
-
-fn generate_compile_success_contract_tests(test_file: &mut File, test_data_dir: &Path) {
-    let test_sub_dir = "compile_success_contract";
-    let test_data_dir = test_data_dir.join(test_sub_dir);
-
-    let test_case_dirs =
-        fs::read_dir(test_data_dir).unwrap().flatten().filter(|c| c.path().is_dir());
+                test_dir = test_dir.display(),
+            )
+            .expect("Could not write templated test file.");
+        });
+
+    let generate_compile_success_contract_code: Box<TestCodeGenerator> =
+        Box::new(|test_file, test_dir, test_name| {
+            write!(
+                test_file,
+                r#"
+#[test]
+fn compile_success_contract_{test_name}() {{
+    let test_program_dir = PathBuf::from("{test_dir}");
 
-    for test_dir in test_case_dirs {
-        let test_name =
-            test_dir.file_name().into_string().expect("Directory can't be converted to string");
-        if test_name.contains('-') {
-            panic!(
-                "Invalid test directory: {test_name}. Cannot include `-`, please convert to `_`"
-            );
-        };
-        let test_dir = &test_dir.path();
+    let mut cmd = Command::cargo_bin("nargo").unwrap();
+    cmd.arg("--program-dir").arg(test_program_dir);
+    cmd.arg("compile");
 
-        write!(
+    cmd.assert().success();
+}}
+            "#,
+                test_dir = test_dir.display(),
+            )
+            .expect("Could not write templated test file.");
+        });
+
+    let generate_compile_failure_code: Box<TestCodeGenerator> = Box::new(
+        |test_file, test_dir, test_name| {
+            write!(
             test_file,
             r#"
 #[test]
-fn compile_success_contract_{test_name}() {{
+fn compile_failure_{test_name}() {{
     let test_program_dir = PathBuf::from("{test_dir}");
 
     let mut cmd = Command::cargo_bin("nargo").unwrap();
     cmd.arg("--program-dir").arg(test_program_dir);
-    cmd.arg("compile");
+    cmd.arg("execute");
 
-    cmd.assert().success();
+    cmd.assert().failure().stderr(predicate::str::contains("The application panicked (crashed).").not());
 }}
             "#,
             test_dir = test_dir.display(),
         )
         .expect("Could not write templated test file.");
-    }
+        },
+    );
+
+    generate_tests(
+        &mut test_file,
+        &test_dir,
+        "execution_success",
+        &*generate_execution_success_code,
+    );
+    generate_tests(
+        &mut test_file,
+        &test_dir,
+        "compile_success_empty",
+        &*generate_compile_success_empty_code,
+    );
+    generate_tests(
+        &mut test_file,
+        &test_dir,
+        "compile_success_contract",
+        &*generate_compile_success_contract_code,
+    );
+    generate_tests(&mut test_file, &test_dir, "compile_failure", &*generate_compile_failure_code);
```

Comment:
```suggestion
    generate_tests(
        &mut test_file,
        &test_dir,
        "execution_success",
        &generate_execution_success_code,
    );
    generate_tests(
        &mut test_file,
        &test_dir,
        "compile_success_empty",
        &generate_compile_success_empty_code,
    );
    generate_tests(
        &mut test_file,
        &test_dir,
        "compile_success_contract",
        &generate_compile_success_contract_code,
    );
    generate_tests(&mut test_file, &test_dir, "compile_failure", &generate_compile_failure_code);
```

These don't need to be dereferenced

Reply:
dang sorry about missing this. Fixed it!